to-do-list-backend/src/api/express.ts

```diff
@@ -54,12 +54,36 @@ export function createServer(dependencies: Dependencies): Express {
         resp.send({});
     })
 
-    app.get('/api/tasks', (_, resp: Response) => {
+    app.get('/api/task', (_, resp: Response) => {
         const tasks = dependencies.storage.listTasks();
 
         resp.status(StatusCodes.OK);
         resp.send({ tasks: tasks })
     })
 
+    app.post('/api/task/mark-done', (req, res) => {
+        const { name } = req.body;
+
+        const task = tasks.find(task => task.name === name);
+        if (task) {
+            task.done = true;
+            res.status(StatusCodes.OK).send(task);
+        } else {
+            res.status(StatusCodes.BAD_REQUEST).send({ error: 'Task not found' });
```

Comment:
It's better to use StatusCodes.NOT_FOUND

The same in /api/task/mark-undone

Reply:
done